client/ayon_maya/plugins/publish/collect_ornatrix_rig.py

```diff
@@ -0,0 +1,80 @@
+import os
+from typing import List, Dict, Any
+import pyblish.api
+from ayon_core.pipeline.publish import KnownPublishError
+from ayon_maya.api import lib
+from ayon_maya.api import plugin
+from maya import cmds
+
+
+ORNATRIX_NODES = {
+    "HairFromGuidesNode", "GuidesFromMeshNode",
+    "MeshFromStrandsNode", "SurfaceCombNode"
+}
+
+
+class CollectOxRig(plugin.MayaInstancePlugin):
+    """Collect all information of the Ornatrix Rig"""
+
+    order = pyblish.api.CollectorOrder + 0.4
+    label = "Collect Ornatrix Rig"
+    families = ["oxrig"]
+
+    def process(self, instance):
+        ornatrix_nodes = cmds.ls(instance.data["setMembers"], long=True)
+        self.log.debug(f"Getting ornatrix nodes: {ornatrix_nodes}")
+
+        ornatrix_resources = []
+        for node in ornatrix_nodes:
+            # Get Yeti resources (textures)
+            resources = self.get_texture_resources(node)
+            ornatrix_resources.extend(resources)
+        # avoid duplicate dictionary data
+        instance.data["resources"] = [
+            i for n, i in enumerate(ornatrix_resources)
+            if i not in ornatrix_resources[n + 1:]
+        ]
+        self.log.debug(instance.data["resources"])
+
+    def get_texture_resources(self, node: str) -> List[Dict[str, Any]]:
+        resources = []
+        node_shape = cmds.listRelatives(node, shapes=True)
+        if not node_shape:
+            return []
```

Comment:
Pretty sure this will never continue.

What nodes are you expecting to put in the instance @moonyuet ? :)

If it's the mesh shape node then it will never have relative child shapes and thus on line 43 this will always `return []`.

I can refactor it to support both in a way - but the first entry point is.

What do we want in the set?

Currently it needs the `cube_GEOShape`:
![image](https://github.com/user-attachments/assets/19b285bf-d652-49fe-bda0-df8ecfe39439)

---

I think either of these make most sense:

The transform of the EditGuidesShape `FurBall1` here:
![image](https://github.com/user-attachments/assets/bf296151-fa7e-40ec-8d0c-7d10b60b84d5)

Or the geometry's transform:
![image](https://github.com/user-attachments/assets/6ec938fc-f726-4428-8754-6ebab6a5a1aa)

Reply:
![image](https://github.com/user-attachments/assets/9f0fdaa1-503f-4b1b-87e8-a1a278d33158)

Reply:
fixed this bug when only happened without texture [18db2be](https://github.com/ynput/ayon-maya/pull/1/commits/18db2be3cdb8086aaf333e92a17a6353187cc697)

Reply:
Yes - works now without texture.